models/new_model.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# DART software - Copyright UCAR. This open source software is provided
+# by UCAR, "as is", without charge, subject to all terms of use at
+# http://www.image.ucar.edu/DAReS/DART/DART_download
+
+function usage { echo "Usage: $0 [model_name] [location_mod]"; }
+
+model_name=$1
+location_mod=$2
+
+if [[ ($# -le 1) ]]; then
+  usage
+  exit 0
+elif [[ -d "$model_name" ]]; then
+  echo "$model_name already exists! Please try a different model name."
+  exit 1
+elif [[ ! -d "../assimilation_code/location/$location_mod" ]]; then
+  echo "$location_mod is not a proper location module! Please try a different location name."
+  exit 1
+fi
+
+mkdir -p "$model_name/work"
+cp "template/new_model.rst" "$model_name/readme.rst"
+sed -i '' "s;template_model;$model_name;" "$model_name/readme.rst"
+cp "template/work/quickbuild.sh" "$model_name/work/quickbuild.sh"
+sed -i '' -e  "s;template_model;$model_name;" -e "s;template_location;$location_mod;" "$model_name/work/quickbuild.sh"
+
+case $location_mod in
+  threed | threed_cartesian | threed_sphere) # threed does not compile, but is a good framework
```

Comment:
The threed version does not compile because the threed location_mod.f90 is out-of-date - the location_mod.f90 should have a get_close_obs and a get_close_state.  
I'll fix that and then I think your script will produce code that compiles for threed.